### PR TITLE
HTML: Add manual accessibility tests for fieldset

### PR DESCRIPTION
**Pull request by zcorpan**

### CODE REVIEW

File: html/semantics/forms/the-fieldset-element/accessibility/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Fieldset accessibility tests
+============================
+
+These tests are intended to test the accessibility of the fieldset and legend elements. They do not
+yet have pass conditions or a clear way to run them. This is expected to be addressed later.
```
Comment (annevk) [inline html/semantics/forms/the-fieldset-element/accessibility/README.md]: This makes them rather hard to review, no?

Comment (zcorpan) [reply html/semantics/forms/the-fieldset-element/accessibility/README.md]: Yeah. :/ I don't know yet what expected behavior should be.

Comment (zcorpan) [reply html/semantics/forms/the-fieldset-element/accessibility/README.md]: Also see https://github.com/w3c/html-aam/issues/145

**Comment by zcorpan** [comment]
If https://github.com/WICG/aom/blob/gh-pages/explainer.md#full-introspection-of-an-accessibility-tree---computedaccessiblenode is implemented, it could be used by these tests as normal testharness.js tests.

**Comment by zcorpan** [comment]
I see `accname/` has some tests for accessible name (but not for fieldset), and although the tests are marked as manual, they have some script to formalize the expected result. I'm not familiar with that. @halindrome @joanmarie should these tests be in `accname/` with that convention? How are those tests run?

**Comment by zcorpan** [comment]
@halindrome @joanmarie ping

**Comment by joanmarie** [comment]
The tests are run via tools created by the ARIA working group which we hope to integrate into WPT, but which are not yet ready for general consumption due to bugs, lack of documentation, etc.

As for where the tests should live, we should get @stevefaulkner and @scottaohara to chime in. The HTML AAM has normative content regarding Accessible Name calculation specific to HTML and which overrides what is in the Acc Name spec.

**Comment by zcorpan** [comment]
@mcking65 are you able to take a look at this?

File: html/semantics/forms/the-fieldset-element/accessibility/aria-manual.html [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<!doctype html>
+<title>fieldset accessibility test: ARIA</title>
+<div role=group aria-labelledby=legend>
+ <div id=legend>Foo</div>
+ <input>
+</div>
+<p>Expected accessible name: "Foo"
```
[the remark this answers is not in the document]
Comment (asurkov) [reply html/semantics/forms/the-fieldset-element/accessibility/aria-manual.html]: I'm also curious why the test are restricted to fieldset accessible name only. Should it have accessible role as well?

File: html/semantics/forms/the-fieldset-element/accessibility/fieldset-div-display-contents-manual.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<!doctype html>
+<title>fieldset accessibility test: fieldset div display: contents</title>
+<style>
+ div { display: contents; }
+</style>
+<fieldset>
+ <div>
+  <legend>Foo</legend>
```
Comment (asurkov) [inline html/semantics/forms/the-fieldset-element/accessibility/fieldset-div-display-contents-manual.html]: According to HTML spec [1], legend has to be a first child of fieldset. So you're right the browsers shouldn't pick up accessible name from a legend, which is not a direct child. I'd be nice to have a comment in the test or extend <title> to reflect this.

[1] https://html.spec.whatwg.org/multipage/form-elements.html#the-legend-element

File: html/semantics/forms/the-fieldset-element/accessibility/legend-child-display-none-manual.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<!doctype html>
+<title>fieldset accessibility test: legend child display: none</title>
+<style>
+ legend > span { display: none; }
```
Comment (asurkov) [inline html/semantics/forms/the-fieldset-element/accessibility/legend-child-display-none-manual.html]: I'd say the test is not legend specific, but it's ok to have it if you want

File: html/semantics/forms/the-fieldset-element/accessibility/legend-role-group-manual.html [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<!doctype html>
+<title>fieldset accessibility test: legend role=group aria-labelledby=fieldset</title>
+<fieldset id=fieldset>
+ <legend role=group aria-labelledby=fieldset>Foo</legend>
```
Comment (asurkov) [inline html/semantics/forms/the-fieldset-element/accessibility/legend-role-group-manual.html]: a tricky one, the output looks reasonable, but I didn't run it through the spec name computation alg

File: html/semantics/forms/the-fieldset-element/accessibility/multiple-legends-manual.html [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<!doctype html>
+<title>fieldset accessibility test: multiple legends</title>
+<fieldset>
+ <legend>Foo</legend>
+ <legend>Bar</legend>
```
Comment (asurkov) [inline html/semantics/forms/the-fieldset-element/accessibility/multiple-legends-manual.html]: a good one, might be worth to have a test where legend if not a first child

File: html/semantics/forms/the-fieldset-element/accessibility/shadow-dom-manual.html [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    <slot name="my-text"></slot>
+    <input>
+  </fieldset>
+</template>
+
+<my-fieldset>
+  <legend slot="my-text">Foo</legend>
+</my-fieldset>
+
+<p>Expected accessible name: ""
```
Comment (asurkov) [inline html/semantics/forms/the-fieldset-element/accessibility/shadow-dom-manual.html]: mm, why?

File: html/semantics/forms/the-fieldset-element/accessibility/legend-visibility-hidden-manual.html [hunk 8]
```diff
@@ -0,0 +1,10 @@
+<!doctype html>
+<title>fieldset accessibility test: legend visibility: hidden</title>
+<style>
+ legend { visibility: hidden; }
```
Comment (asurkov) [inline html/semantics/forms/the-fieldset-element/accessibility/legend-visibility-hidden-manual.html]: might be worth to have display:none/visibility:hidden placed on fieldset element. Also might worth testing visibility:collapsed and visibility:visible

**Review by asurkov — APPROVED** [review]
overall it looks good, sorry it took that long

File: html/semantics/forms/the-fieldset-element/accessibility/aria-manual.html [hunk 9]
```diff
@@ -0,0 +1,7 @@
+<!doctype html>
+<title>fieldset accessibility test: ARIA</title>
+<div role=group aria-labelledby=legend>
```
[the remark this answers is not in the document]
Comment (zcorpan) [reply html/semantics/forms/the-fieldset-element/accessibility/aria-manual.html]: We don't have a style guide, and unquoted attributes certainly are used elsewhere in wpt.

File: html/semantics/forms/the-fieldset-element/accessibility/fieldset-div-display-contents-manual.html (hunk 3 above)
Comment (zcorpan) [reply html/semantics/forms/the-fieldset-element/accessibility/fieldset-div-display-contents-manual.html]: That is an authoring conformance requirement. The UA conformance requirement is

https://w3c.github.io/html-aam/#fieldset-element-accessible-name-computation

step 2

File: html/semantics/forms/the-fieldset-element/accessibility/legend-role-group-manual.html (hunk 5 above)
Comment (zcorpan) [reply html/semantics/forms/the-fieldset-element/accessibility/legend-role-group-manual.html]: Yeah the spec side is pretty messy. I filed an issue but can't find it now. Basically I think accname should have hooks for html-aam to use when defining fieldset/legend.

File: html/semantics/forms/the-fieldset-element/accessibility/shadow-dom-manual.html (hunk 7 above)
Comment (zcorpan) [reply html/semantics/forms/the-fieldset-element/accessibility/shadow-dom-manual.html]: Accessible name computation works on the DOM tree, not the flat tree.

**Comment by zcorpan** [comment]
I can't find the spec for accessible name and descriptions anymore... https://github.com/w3c/accname/issues/34

**Comment by zcorpan** [comment]
OK addressed comments, thank you @asurkov

**Review by sideshowbarker — APPROVED** [review]

**Comment by zcorpan** [comment]
Rebased on latest master. Should be good to merge if bots are green.

**Review by sideshowbarker — APPROVED** [review]

**Comment by zcorpan** [comment]
Thanks Mike!

**Comment by sideshowbarker** [comment]
Cheers 🍫

**Pull request merged**